Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp

```diff
@@ -211,6 +302,78 @@ void NetKDRequestDevice::LogError(ErrorType error_type, s32 error_code)
   m_scheduler_buffer[2] = Common::swap32(new_code);
 }
 
+NWC24::ErrorCode NetKDRequestDevice::KDCheckMail(u32* _mail_flag, u32* _interval)
+{
+  m_scheduler_buffer[4] = Common::swap32(2);
+  u64 random_number{};
+  Common::Random::Generate(&random_number, sizeof(u64));
+  const std::string form_data(
+      fmt::format("mlchkid={}&chlng={}", m_config.GetMlchkid(), random_number));
+  const Common::HttpRequest::Response response = m_http.Post(m_config.GetCheckURL(), form_data);
+
+  if (!response)
+  {
+    ERROR_LOG_FMT(IOS_WC24, "NET_KD_REQ: IOCTL_NWC24_CHECK_MAIL_NOW: Failed to request data at {}.",
+                  m_config.GetCheckURL());
+    return NWC24::WC24_ERR_SERVER;
+  }
+
+  const std::string response_str = {response->begin(), response->end()};
+  const std::string code = GetValueFromCGIResponse(response_str, "cd");
+  if (code != "100")
+  {
+    ERROR_LOG_FMT(
+        IOS_WC24,
+        "NET_KD_REQ: IOCTL_NWC24_CHECK_MAIL_NOW: Mail server returned non-success code: {}", code);
+    return NWC24::WC24_ERR_SERVER;
+  }
+
+  const std::string server_hmac = GetValueFromCGIResponse(response_str, "res");
+  const std::string mail_flag = GetValueFromCGIResponse(response_str, "mail.flag");
+  const std::string interval = GetValueFromCGIResponse(response_str, "interval");
+  DEBUG_LOG_FMT(IOS_WC24, "NET_KD_REQ: IOCTL_NWC24_CHECK_MAIL_NOW: Server HMAC: {}", server_hmac);
+
+  // On a real Wii, a response to a challenge is excepted and would be verified by KD.
```

Comment:
This comment ready a bit weird in my head; at least the word "excepted" seems off. Accepted? Exempt? Expected? Something else?

Thinking "expected" but I'm not sure if thats what you had in mind.

Reply:
Was supposed to be `expected`, thanks for catching that